desc/objectives/_geometry.py

```diff
@@ -707,22 +728,265 @@ def compute(self, equil_params, surface_params=None, constants=None):
         )
         plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
         if self._surface_fixed:
-            surface_coords = constants["surface_coords"]
+            data_surf = constants["data_surf"]
         else:
-            surface_coords = compute_fun(
+            data_surf = compute_fun(
                 self._surface,
                 self._surface_data_keys,
                 params=surface_params,
                 transforms=constants["surface_transforms"],
                 profiles={},
                 basis="xyz",
-            )["x"]
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+            )
+
+        surface_coords = data_surf["x"]
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
 
         if self._use_softmin:  # do softmin
             return jnp.apply_along_axis(softmin, 0, d, self._alpha)
         else:  # do hardmin
-            return d.min(axis=0)
+            if not self._use_signed_distance:
+                return d.min(axis=0)
+            else:
+                min_inds = d.argmin(axis=0, keepdims=True)
+                min_ds = jnp.take_along_axis(d, min_inds, axis=0).squeeze()
+                diff_vecs_at_mins = jnp.take_along_axis(
+                    diff_vec, min_inds[..., None], axis=0
+                ).squeeze()
+                n_rho_dot_diff_vec = dot(
+                    data_surf["n_rho"], diff_vecs_at_mins, axis=-1
+                ).squeeze()
+                # we mult by -1 because diff_vec points from surface to
+                # the plasma, so
+                # if plasma is inside the surface, n_rho dot diff vec <0,
+                # but we define "plasma inside surface" as >0
+                return min_ds * jnp.sign(n_rho_dot_diff_vec) * -1
+
+
+class PlasmaVesselDistanceCircular(_Objective):
```

Comment:
How much more efficient/effective is this that just using the regular one? Do we need a separate objective just for this case? It seems like this might be specific to NTtau stuff.

Reply:
It is more effective in that it definitely assigns signs correctly, but yea it is a bit constrained to the NT tao stuff.

Reply:
I agree this is limited in application to small-scale experiments, like nTtau and Muse. It might not be worth maintaining an extra objective function for such a niche use case, if the original one works fine.